### ISSUE TITLE
Authorization headers not send in IE11

### ISSUE THREAD

**Issue by peteruithoven** [micky2be/superlogin-client#40]
Superlogin-client has a [checkEndpoint](https://github.com/micky2be/superlogin-client/blob/master/src/index.js#L17-L26) function which checks whether the authorization headers should be send with a request to a certain endpoint. The current host for example is always included. 

It looks like this check always fails on IE11 (and probably the older IE's as well). 

The [checkEndpoint](https://github.com/micky2be/superlogin-client/blob/master/src/index.js#L17-L26) function uses a `a` element to parse the url of a request. It gives the url as `href` and it then checks the `host` properly. When giving `https://example.com/users` as href, host should return: `example.com`. 
I can't find this `host` property in [Mozilla's A element documentation](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a) and on IE 11 it seems to return return nothing (not sure whether it returns undefined / null or an empty string yet). Strangly enough when I try the same in the Console of IE11 it does work, maybe there's a race condition involved. 

But anyway, isn't there a better / saver way to parse a url than using a html element?

**Comment by peteruithoven** [comment]
This one seems interesting: http://stackoverflow.com/a/13405933/1544200
> IE doesn't populate all link properties when setting .href with a relative URL

So in IE:
1. `/api/users` > host: `''`
2. `example.com/api/users` > host: `''`
3. `http://example.com/api/users` > host: `'example.com:80'

Also what about those port numbers? The documentation doesn't explain these need to be added to the `endpoints` config.

**Comment by peteruithoven** [comment]
Case 2 is actually a weird example, it's basically a invalid url, that shouldn't be used anyway. 
Case 3 could be solved by using `window.location.host` when `parser.host` is empty. 

We could also use a url parsing library like: [url-parse](https://www.npmjs.com/package/url-parse). Recent versions don't use a A html element anymore to make it compatible with environments like webworkers. (Also it has a high score on npms.io: [https://npms.io/search?q=url](https://npms.io/search?q=url))

**Comment by micky2be** [comment]
You should just use absolute url. It will fix your problem.

**Comment by micky2be** [comment]
> But anyway, isn't there a better / saver way to parse a url than using a html element?

Using the a HTML tag is the easiest and best way to parse a url on the front. Everything is built in.

> Also what about those port numbers? The documentation doesn't explain these need to be added to the endpoints config.

That's what an endpoint is, host+port+path. I'm not gonna put the definition of endpoint in the documentation.

> We could also use a url parsing library like: url-parse.

I don't really mind using a library but I would rather use the Node counterpart https://www.npmjs.com/package/url which will also require a full url.

**Comment by peteruithoven** [comment]
> You should just use absolute url. It will fix your problem.

We're running our current project in 3 places (local, dev server and stable server), we don't currently have a way to retrieve env variables in client side javascript, so getting absolute url's is problematic. 



>> Also what about those port numbers? The documentation doesn't explain these need to be added to the endpoints config.

> That's what an endpoint is, host+port+path. I'm not gonna put the definition of endpoint in the documentation.

But looking at the readme:
```
  // A list of API endpoints to automatically add the Authorization header to
  // By default the host the browser is pointed to will be added automatically
  endpoints: ['api.example.com'],
```
You don't actually mention you only compare those with the host of the request? (`if (host === endpoints[i])`). Where do you check the path? 
And if you just compare with the host (which always seems to includes a port, how can it ever equal `api.example.com`? Shouldn't it be `api.example.com:433`?

> I don't really mind using a library but I would rather use the Node counterpart https://www.npmjs.com/package/url which will also require a full url.

But why? It has, less releases, last updated longer ago, is still a 0. release, it still has issues from 2014 that didn't even get a label. It might not be up to date with node's latest spec (https://github.com/defunctzombie/node-url/issues/24)? It is however twice as popular I see.

**Comment by micky2be** [comment]
Using full url guaranty where you are sending data. A bit more secure.
I'll update the Readme, and yes we should technically check the path (legacy from Angular version)

I do use multiple environment as well, I recommend you to use a system to inject variable when building your frontend.
You can use node-config to set different config based on environment for example.
From there using webpack, gulp, or others, you should be able to do some magic ;)

**Comment by polco** [comment]
there is one big problem in using dom elements: we cannot use superlogin inside a web worker. the window.localStore / window.sessionStorage also breaks.
(even though that was not the problem in that issue :) )

**Comment by peteruithoven** [comment]
@polco that's also an issue in: https://github.com/micky2be/superlogin-client/issues/35

**Comment by micky2be** [comment]
@peteruithoven If you could make a PR that would be highly appreciated.
Use the URL parser of your choice.

But again, I recommend you to use full URL for your application.

**Comment by micky2be** [comment]
Fixed in #53
